src/pages/iou/IOUModal.js

```diff
@@ -286,7 +286,24 @@ class IOUModal extends Component {
         const amount = Math.round(this.state.amount * 100);
         const currency = this.props.iou.selectedCurrencyCode;
         const comment = this.state.comment;
+        const chatReportID = lodashGet(this.props, 'route.params.reportID', '');
+
+        // If it's paying with the wallet lets call the correct service for this
+        if (paymentMethodType === CONST.IOU.PAYMENT_TYPE.EXPENSIFY) {
+            IOU.sendMoneyWithWallet(
+                chatReportID,
+                amount,
+                currency,
+                comment,
+                this.state.participants,
+            );
+
+            // Exit to the report as the API pusher response will be handled elsewhere
+            Navigation.navigate(ROUTES.getReportRoute(chatReportID));
+            return;
+        }
 
+        // This should be default call while we are doing the refactor
```

Comment:
This should be deprecated/ removed, we cannot make two calls, not sure what you mean here, we would pay twice 😅

Reply:
no not pay twice, I meant once Youssef does his refactor this would be removed, I already return ealry if the payment type is Wallet, so it won't do it twice

Reply:
Ah I see, I would remove it here just so we can test, but I see you point now, thanks for explaining